### PR TITLE
doc: CChainState return values

### PR DESCRIPTION
**Pull request by maflcko**
It will always return true, unless a system error such as #15305 occurred

### CODE REVIEW

File: src/validation.h [hunk 1]
```diff
@@ -211,7 +211,7 @@ static const uint64_t MIN_DISK_SPACE_FOR_BLOCK_FILES = 550 * 1024 * 1024;
  * @param[in]   pblock  The block we want to process.
  * @param[in]   fForceProcessing Process this block even if unrequested; used for non-network block sources and whitelisted peers.
  * @param[out]  fNewBlock A boolean which is set to indicate if the block was first received via this call
- * @return True if state.IsValid()
+ * @returns If the block was processed. (Might return true for invalid blocks or false for valid blocks)
```
Comment (promag) [inline src/validation.h]: 2nd part is strange, maybe `@returns If the block was processed independently of block validity`.

Comment (MarcoFalke) [reply src/validation.h]: Fixed

File: src/validation.cpp [hunk 2]
```diff
@@ -2472,6 +2472,8 @@ void CChainState::PruneBlockIndexCandidates() {
 /**
  * Try to make some progress towards making pindexMostWork the active block.
  * pblock is either nullptr or a pointer to a CBlock corresponding to pindexMostWork.
+ *
+ * @returns true unless a system error occurred
  */
 bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainParams& chainparams, CBlockIndex* pindexMostWork, const std::shared_ptr<const CBlock>& pblock, bool& fInvalidFound, ConnectTrace& connectTrace)
```
Comment (laanwj) [inline src/validation.cpp]: As the method is defined there, shouldn't this documentation be in the `.h` file? (same for `CChainState::ActivateBestChain` below)

Comment (MarcoFalke) [reply src/validation.cpp]: `ActivateBestChainStep` is marked `private` and was never meant to be exposed in the header. I think it makes sense to keep the documentation close to the function implementation

Comment (MarcoFalke) [reply src/validation.cpp]: Good point for the `public` method, though. So I fixed this for ABC only

**Comment by MarcoFalke** [comment]
Added a commit that can be reviewed with the git diff options `--color-moved=dimmed-zebra --color-moved-ws=ignore-all-space`.

**Comment by laanwj** [comment]
ACK fa912a8ad5a94cd2bdc149400b1befb346621f03

**Pull request merged**